openmc/data/multipole.py

```diff
@@ -0,0 +1,628 @@
+        if l_value is not None:
+            cv.check_type('l_value', l_value, np.ndarray)
+            if len(l_value.shape) != 1:
+                raise ValueError('Multipole l_value arrays must be 1D')
+            if not np.issubdtype(l_value.dtype, np.integer):
+                raise TypeError('Multipole l_value arrays must be integer'
+                                ' dtype')
+        self._l_value = l_value
+
+    @w_start.setter
+    def w_start(self, w_start):
+        if w_start is not None:
+            cv.check_type('w_start', w_start, np.ndarray)
+            if len(w_start.shape) != 1:
+                raise ValueError('Multipole w_start arrays must be 1D')
+            if not np.issubdtype(w_start.dtype, np.integer):
+                raise TypeError('Multipole w_start arrays must be integer'
+                                ' dtype')
+        self._w_start = w_start
+
+    @w_end.setter
+    def w_end(self, w_end):
+        if w_end is not None:
+            cv.check_type('w_end', w_end, np.ndarray)
+            if len(w_end.shape) != 1:
+                raise ValueError('Multipole w_end arrays must be 1D')
+            if not np.issubdtype(w_end.dtype, np.integer):
+                raise TypeError('Multipole w_end arrays must be integer dtype')
+        self._w_end = w_end
+
+    @broaden_poly.setter
+    def broaden_poly(self, broaden_poly):
+        if broaden_poly is not None:
+            cv.check_type('broaden_poly', broaden_poly, np.ndarray)
+            if len(broaden_poly.shape) != 1:
+                raise ValueError('Multipole broaden_poly arrays must be 1D')
+            if not np.issubdtype(broaden_poly.dtype, np.bool):
+                raise TypeError('Multipole broaden_poly arrays must be boolean'
+                                ' dtype')
+        self._broaden_poly = broaden_poly
+
+    @curvefit.setter
+    def curvefit(self, curvefit):
+        if curvefit is not None:
+            cv.check_type('curvefit', curvefit, np.ndarray)
+            if len(curvefit.shape) != 3:
+                raise ValueError('Multipole curvefit arrays must be 3D')
+            if curvefit.shape[2] != 3:  # One each for sigT, sigA, sigF
+                raise ValueError('The third dimension of multipole curvefit'
+                                 ' arrays must have a length of 3')
+            if not np.issubdtype(curvefit.dtype, np.float):
+                raise TypeError('Multipole curvefit arrays must be float dtype')
+        self._curvefit = curvefit
+
+    @classmethod
+    def from_hdf5(cls, group_or_filename):
+        if isinstance(group_or_filename, h5py.Group):
+            group = group_or_filename
+        else:
+            h5file = h5py.File(group_or_filename, 'r')
+            version = h5file['version'].value
+            if version != WMP_VERSION:
+                raise ValueError('The given WMP data uses version '
+                    + str(version) + ' whereas your installation of the OpenMC '
```

Comment:
Looks like the version dataset is written as an array rather than a scalar, so you'd need to do `version[0].decode()`. If you did `str(version)`, you'd actually get `[b'v0.2']`.